packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt

```diff
@@ -200,6 +215,84 @@ class DatadogSdkPlugin : FlutterPlugin, MethodCallHandler {
         isRegistered.set(false)
     }
 
+    @Suppress("TooGenericExceptionCaught")
+    private fun mapLogEvent(event: LogEvent): LogEvent? {
+        val jsonEvent = event.toJson().asMap()
+
+        var modifiedJson: Map<String, Any?>? = null
+
+        val semaphore = Semaphore(1)
+        semaphore.acquire()
+
+        val handler = Handler(Looper.getMainLooper())
+        handler.post {
+            try {
+                channel.invokeMethod(
+                    "mapLogEvent",
+                    mapOf(
+                        "event" to jsonEvent
+                    ),
+                    object : MethodChannel.Result {
+                        override fun success(result: Any?) {
+                            modifiedJson = result as? Map<String, Any?>
+                            semaphore.release()
+                        }
+
+                        override fun error(
+                            errorCode: String,
+                            errorMessage: String?,
+                            errorDetails: Any?
+                        ) {
+                            // No telemetry needed, this is likely an issue in user code
+                            semaphore.release()
+                        }
+
+                        override fun notImplemented() {
+                            Datadog._internal._telemetry.error(
+                                "mapLogEvent returned notImplemented."
+                            )
+                            semaphore.release()
+                        }
+                    }
+                )
+            } catch (t: Throwable) {
+                Datadog._internal._telemetry.error("Attempting call mapLogEvent failed.", t)
+                semaphore.release()
+            }
+        }
+        // Stalls until the method channel finishes
+        semaphore.acquire()
+
+        try {
+            var modifiedEvent: LogEvent? = null
+            modifiedJson?.let {
+                modifiedEvent = LogEvent.fromJsonObject(it.toJsonObject())
+            }
```

Comment:
minor: a bit more clean imo, but this is opinionated (benefits from smart cast + val vs var):

```suggestion
            val modifiedEvent = if (modifiedJson != null) {
                LogEvent.fromJsonObject(modifiedJson.toJsonObject())
            } else {
                null
            }
```